server/src/main/java/org/candlepin/bind/DelayedPoolOp.java

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright (c) 2009 - 2017 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.bind;
+
+import org.candlepin.controller.PoolManager;
+import org.candlepin.model.Entitlement;
+import org.candlepin.model.Pool;
+import org.candlepin.model.SourceSubscription;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+/**
+ * Holds and represents operations to be performed at a later time.
+ * Currently, only holds relevant operations for PostBindBonusPoolsOp
+ */
+public class DelayedPoolOp {
```

Comment:
I'd prefer a different name here.  This class doesn't implement `BindOperation` so ending it with `Op` is kind of confusion.  Could I suggest `BindCallback` or `PoolCallback` or something like that?

Reply:
@awood we do have a precedence in RevocationOp that does end with Op and does not implement BindOperation. Going forward I can see us having similar Op classes for refresh code that will not implement BindOperation, but might implement something like RefreshOp ....

that being said, if you hate the name I am not married to it.. any objection to PoolOperationCallback ?

Reply:
PoolOperationCallback works for me!